### DISCUSSION TITLE
Logik für Namensgebung

### DISCUSSION THREAD

**Discussion by patman15** [SukramJ/aiohomematic#609] [Q&A] [answered]
Hallo!

Ich habe gesehen, die Namensgebung wurde schon mehrfach diskutiert und ich will sie auch gar nicht hinterfragen sondern nur besser verstehen, damit ich weiß nach welchem Schema ich vorgehen sollte um nachträglich nicht Probleme zu bekommen.

Wenn ich richtig gesehen habe, dann wird
- die `entity_id` beim ersten Start auf Grund des Namens in HM angelegt und
- im Falle einer Änderung in HM wird ein `friendly_name` Attribut hinzugefügt

Stimmt das soweit?
Falls ja, sollte ich alle Namen in HM vor dem ersten Start sauber haben, damit ich nachher in diversen Konfigurationen nicht unlogische `entity_id`, z.B. in template sensors stehen habe, oder gibt's da eine bessere Methode?
Ist es eine gute Idee den JSON-RPC Port erst nach dem ersten Start anzugeben, damit die `entity_id` alle mit den (zugegeben hässlichen) Seriennummern angelegt werden, dafür dann aber auf ewig konstant sind? (Hab recht viele template Sensoren und YAML Dateien wo ich oft die `entity_id` angeben muss.)

Wäre da für gute Ideen dankbar, da ich gerade evaluiere, was ich alles wann umstellen muss um von der alten Integration endlich zu der coolen Neuen zu kommen. Danke!

**Comment by SukramJ** [comment, marked as answer]
> Stimmt das soweit?

Ja.

> Ist es eine gute Idee den JSON-RPC Port 

Das macht keinen Sinn, da der JSON-RPC immer verwendet wird. Hier kann man nur einen abweichenden Port (Std: 80/443) angeben.

**Reply by patman15** [reply]
> Das macht keinen Sinn, da der JSON-RPC immer verwendet wird. Hier kann man nur einen abweichenden Port (Std: 80/443) angeben.

Ich könnte einen falschen angeben und dann ändern ...

Wie auch immer, also kurz, man sollte in der HM ein konsistentes Namensschema haben, dass sich nicht mehr ändert, sonst läuft man in Probleme (auf die `entity_id` bezogen, weil die GUI arbeitet eh mit `friendly_name`), ist das die Intention?

**Reply by SukramJ** [reply]
> Ich könnte einen falschen angeben und dann ändern ...

Mach bei Problemen kein Ticket auf!


Wie Du selber schreibst:

>   die entity_id beim ersten Start auf Grund des Namens in HM angelegt 

Die entity_id ändert sich, ausser ggf. bei einer Neuinstallion nicht mehr, kann also in Automationen usw. verwendet werden.
D.h. Änderungen in der CCU spiegeln sich nachträglich nicht in der entity_id wieder.

Zur Anzeige in der UI wird der `friendly_name` verwendet.

Sowohl `entity_id` als auch `friendly_name` können in der UI überschrieben werden.

**Reply by SukramJ** [reply]
Und [das](https://github.com/danielperna84/hahomematic/wiki/Howto's#rename-of-device-in-ccu-not-reflected-in-home-assistant) geht auch noch.

**Reply by patman15** [reply]
> Mach bei Problemen kein Ticket auf!

Versprochen! :-)
Danke!

**Comment by emufan** [comment]
> damit die `entity_id` alle mit den (zugegeben hässlichen) Seriennummern angelegt werden, dafür dann aber auf ewig konstant sind?

Ewig konstant, bis Gerät kaputt und ersetzt werden muss. Daher mache ich das Gegenteil. Strukturierte Namen geben. Und wenn ein Gerät ersetzt werden muss, dann bekommt es die gleiche `` entity_id` wie das alte (was vorher gelöscht wird) und Templates, Automatisierungen, usw. laufen weiter wie vorher.

**Reply by ThomDietrich** [reply]
Stimme hier voll zu, möglichst **erst** saubere Namen in HM, dann die Verbindung zu HA.

Die Struktur sollte dabei gut überlegt sein. @emufan wie machst du es? Hier mein Schema. Es stehen viele gut gemeinte Ideen dahinter aber manche Sonderfälle sind auch bei mir nicht konsequent gelöst:

![image](https://user-images.githubusercontent.com/2870104/187465595-d459a0b7-96e5-4ef7-a9c4-8ff4b98e8b08.png)

**Reply by emufan** [reply]
Ähnlich, aber andersrum. Erst Gewerk und dann Raum. Heizung Wohnzimmer, Heizung Bad, usw. Rolladen Wohnzimmer hinten rechts, ...Historisch gewachsen, anders rum, d.h. wie bei Dir ist bei mehreren Aktoren, insb. Lichtern und Tastern je Raum sinniger, aber da habe ich nie umgestellt und diese haben oft auch eindeutige Namen, wo ich keinen Raum im Namen habe, damit es sich per Sprache leichter bedienen lässt, Schreibtischlampe, usw.

**Comment by patman15** [comment]
Thanks @SukramJ und @emufan dadurch ist mir einiges klarer geworden.
